dnsflow_read.py

```diff
@@ -1,125 +1,179 @@
 #!/usr/bin/env python
 
-'''
+"""
 See dnsflow.c header comment for packet formats.
-'''
 
-import sys, time, argparse
+Utility functions to simplify interface.
+"""
+import time
+import argparse
+import gzip
 import socket
-import dpkt, pcap
+import dpkt
+import pcap
```

Comment:
Thanks for calling this out. What about adding an assertion to check that the right `pcap` module is loaded. If not, raise an exception stating that `pypcap` needs to be installed and you may have to uninstall `python-libpcap` that is leftover from older installs.

Reply:
This is a good suggestion. I was going to add a note about this in the wiki here https://deepfield.atlassian.net/wiki/spaces/SE/pages/230031865/How+To+setup+a+basic+DNSflow+server+and+test, but it would be good to have the code itself fail with a very specific error message incase that note gets missed. I'll add it after lunch.